src/hypervisor-integration.adoc

```diff
@@ -0,0 +1,153 @@
+[#section_hypervisor_integration]
+== Integrating Zcheripurecap and Zcherilegacy with the Hypervisor Extension
+
+The RISC-V hypervisor (H) extension virtualizes the supervisor-level
+architecture to support the efficient hosting of guest operating systems atop a
+type-1 or type-2 hypervisor cite:[riscv-priv-spec]. The hypervisor extension is
+generally orthogonal to CHERI; the main requirements, when integrating with
+Zcheripurecap and Zcherilegacy, is that address CSRs added for hypervisors are
+extended to CLEN size so that they are able to hold capabilities.
+
+[#hstatus,reftext="hstatus"]
+=== Hypervisor Status Register (hstatus)
+
+The <<hstatus>> register operates as described in cite:[riscv-priv-spec] except
+for the VSXL field that controls the value of XLEN for VS-mode (known as
+VSXLEN).
+
+The encoding of the VSXL field is the same as the MXL field of **misa**. Only 1
+and 2 are supported values for VSXL. When the implementation supports
+{cheri_base_ext_name} (but not {cheri_legacy_ext_name}), then <<hstatus>>'s
+VSXL must be read-only as described in xref:mstatus[xrefstyle=short] for
+<<mstatus>>.SXL. When the implementation supports both {cheri_base_ext_name}
+and {cheri_legacy_ext_name}, then VSXL behaves as described in
+xref:status_legacy[xrefstyle=short] for mstatus.SXL.
+
+[#henvcfg,reftext="henvcfg"]
+=== Hypervisor Environment Configuration Register (henvcfg)
+
+The <<henvcfg>> register operates as described in the RISC-V Privileged
+Specification. A new enable bit is added to <<henvcfg>> when the implementation
+supports {cheri_legacy_ext_name} as shown in xref:henvcfgreg[xrefstyle=short].
+
+.Hypervisor environment configuration register (*henvcfg*)
+[#henvcfgreg]
+include::img/henvcfgreg.edn[]
+
+The CHERI Register Enable (<<CRE>>) bit controls whether explicit access to
+CHERI register is permitted when V=1. When <<CRE>>=1, CHERI registers
+can be read and written by VS and VU modes. When <<CRE>>=0, CHERI registers are
+disabled in VS and VU modes.
```

Comment:
Presumably the intent is that the less privileged modes all read their CRE bits as 0 as well? I.e. henvcfg.cre gets added to the cross-CSR wall field?

Reply:
This was recently changed and the CRE bits are no longer a WARL